### ISSUE TITLE
MySQL + Stateful ingestion config/checkpoint bug

### ISSUE THREAD

**Issue by jeremypharo** [datahub-project/datahub#6382]
# Description: MySQL + Stateful ingestion bug
**tl;dr;** Not having luck with statefulness for MySQL, and I think it has to do with some config changes in 8.45 (TwoTierSQLAlchemySource?). The MySQL recipe(s) in question have been running successfully for many months--this was our first attempt at statefulness in MySQL (and we have used statefulness successfully for BigQuery, I should note)

**Datahub Version:**
DataHub CLI version: 0.8.45.1

## Steps To Reproduce
For a MySQL source
1. Enabled stateful ingestion
2. Run (first) ingestion
3. Deleted a couple of tables in the "dev" database we connect to when testing
4. Re-ran ingestion
   * then we get some errors at certain points during the ingestion. The ingestion still finishes successfully aside from the statefulness part.

## Expected behavior
I would have expected during the 2nd run to see the deleted table(s) should up in the `soft_deleted_stale_entities` entry in the source report, and also to not see the aforementioned config error

## Behavior seen
Even the first stateful ingestion run sends over both `schema_pattern` and `database_pattern` as they are both part of the config that gets serialized and sent to the API as the checkpoint data, so when we rerun stateful ingestion, the config object blows up saying we can't have both of those keys--doesn't seem to be a way as a consumer to code around this issue.

https://github.com/datahub-project/datahub/blob/9a687caae4042c379fff4b37ca3774c47b02669b/metadata-ingestion/src/datahub/ingestion/source/sql/two_tier_sql_source.py#L21-L25 

https://github.com/datahub-project/datahub/blob/9a687caae4042c379fff4b37ca3774c47b02669b/metadata-ingestion/src/datahub/configuration/common.py#L130-L150

https://github.com/datahub-project/datahub/blob/9a687caae4042c379fff4b37ca3774c47b02669b/metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/DatahubIngestionCheckpoint.pdl#L30-L33

**datahub --debug ingest output (note schema_pattern and database_pattern both being sent**
```
[2022-11-08 14:02:25,512] INFO     {datahub.ingestion.run.pipeline:433} - Processing commit request for DatahubIngestionCheckpointingProvider. Commit policy = CommitPolicy.ON_NO_ERRORS, has_errors=False, has_warnings=False
[2022-11-08 14:02:25,512] INFO     {datahub.ingestion.source.state_provider.datahub_ingestion_checkpointing_provider:141} - Committing ingestion checkpoint for pipeline:'mysql_dev_etsy_giftcards_pipeline',instance:'mysql_devdbgifts001a.c.etsy-mysql-dev.internal:3306_giftcards', job:'mysql_stale_entity_removal'
[2022-11-08 14:02:25,512] DEBUG    {datahub.emitter.rest_emitter:249} - Attempting to emit to DataHub GMS; using curl equivalent to:
curl -X POST -H 'User-Agent: python-requests/2.28.1' -H 'Accept-Encoding: gzip, deflate' -H 'Accept: */*' -H 'Connection: keep-alive' -H 'X-RestLi-Protocol-Version: 2.0.0' -H 'Content-Type: application/json' --data '{"proposal": {"entityType": "dataJob", "entityUrn": "urn:li:dataJob:(urn:li:dataFlow:(datahub,[REDACTED],prod),mysql_stale_entity_removal)", "changeType": "UPSERT", "aspectName": "datahubIngestionCheckpoint", "aspect": {"value": "{\"timestampMillis\": 1667952139035, \"partitionSpec\": {\"type\": \"FULL_TABLE\", \"partition\": \"FULL_TABLE_SNAPSHOT\"}, \"pipelineName\": \"mysql_dev_etsy_giftcards_pipeline\", \"platformInstanceId\": \"[REDACTED]\", \"config\": \"{\\\"env\\\": \\\"PROD\\\", \\\"platform\\\": null, \\\"platform_instance\\\": null, \\\"stateful_ingestion\\\": {\\\"enabled\\\": true, \\\"max_checkpoint_state_size\\\": 16777216, \\\"state_provider\\\": {\\\"type\\\": \\\"datahub\\\", \\\"config\\\": null}, \\\"ignore_old_state\\\": false, \\\"ignore_new_state\\\": false, \\\"remove_stale_metadata\\\": true, \\\"fail_safe_threshold\\\": 95.0}, \\\"options\\\": {\\\"echo\\\": true}, \\\"schema_pattern\\\": {\\\"allow\\\": [\\\".*\\\"], \\\"deny\\\": [], \\\"ignoreCase\\\": true}, \\\"table_pattern\\\": {\\\"allow\\\": [\\\".*\\\"], \\\"deny\\\": [\\\".*__ptosc.*\\\"], \\\"ignoreCase\\\": true}, \\\"view_pattern\\\": {\\\"allow\\\": [\\\".*\\\"], \\\"deny\\\": [\\\".*__ptosc.*\\\"], \\\"ignoreCase\\\": true}, \\\"profile_pattern\\\": {\\\"allow\\\": [\\\".*\\\"], \\\"deny\\\": [], \\\"ignoreCase\\\": true}, \\\"domain\\\": {}, \\\"include_views\\\": true, \\\"include_tables\\\": true, \\\"profiling\\\": {\\\"enabled\\\": false, \\\"limit\\\": null, \\\"offset\\\": null, \\\"report_dropped_profiles\\\": false, \\\"turn_off_expensive_profiling_metrics\\\": false, \\\"profile_table_level_only\\\": false, \\\"include_field_null_count\\\": true, \\\"include_field_min_value\\\": true, \\\"include_field_max_value\\\": true, \\\"include_field_mean_value\\\": true, \\\"include_field_median_value\\\": true, \\\"include_field_stddev_value\\\": true, \\\"include_field_quantiles\\\": false, \\\"include_field_distinct_value_frequencies\\\": false, \\\"include_field_histogram\\\": false, \\\"include_field_sample_values\\\": true, \\\"max_number_of_fields_to_profile\\\": null, \\\"profile_if_updated_since_days\\\": 1, \\\"profile_table_size_limit\\\": 1, \\\"profile_table_row_limit\\\": 50000, \\\"max_workers\\\": 40, \\\"query_combiner_enabled\\\": true, \\\"catch_exceptions\\\": true, \\\"partition_profiling_enabled\\\": true, \\\"bigquery_temp_table_schema\\\": null, \\\"partition_datetime\\\": null}, \\\"username\\\": \\\"datahub-ro-dev-001\\\", \\\"host_port\\\": \\\"[REDACTED]:3306\\\", \\\"database\\\": \\\"giftcards\\\", \\\"database_alias\\\": null, \\\"scheme\\\": \\\"mysql+pymysql\\\", \\\"sqlalchemy_uri\\\": null, \\\"database_pattern\\\": {\\\"allow\\\": [\\\"giftcards\\\"], \\\"deny\\\": [], \\\"ignoreCase\\\": true}}\", \"state\": {\"formatVersion\": \"1.0\", \"serde\": \"base85\", \"payload\": \"[REDACTED]"}, \"runId\": \"mysql-2022_11_08-14_01_59\"}", "contentType": "application/json"}}}' 'http://ddr-dev.datahub-gms.[REDACTED]/aspects?action=ingestProposal'
\[2022-11-08 14:02:25,785] INFO     {datahub.ingestion.source.state_provider.datahub_ingestion_checkpointing_provider:167} - Committed ingestion checkpoint for pipeline:'mysql_dev_etsy_giftcards_pipeline',instance:'[REDACTED]', job:'mysql_stale_entity_removal'
[2022-11-08 14:02:25,785] INFO     {datahub.ingestion.run.pipeline:452} - Successfully committed changes for DatahubIngestionCheckpointingProvider.
[2022-11-08 14:02:25,826] INFO     {datahub.cli.ingest_cli:150} - Finished metadata ingestion
[2022-11-08 14:02:25,826] DEBUG    {datahub.telemetry.telemetry:243} - Sending Telemetry
```

### Error message

```
[2022-11-04 13:01:01,082] INFO     {datahub.ingestion.source.state_provider.datahub_ingestion_checkpointing_provider:94} - The last committed ingestion checkpoint for pipelineName:'mysql_dev_pipeline', platformInstanceId:'[REDACTED]', job_name:'mysql_stale_entity_removal' found with start_time: 2022-11-04 20:57:14.112000+00:00 and a bucket duration of None.
--- Logging error ---
Traceback (most recent call last):
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/state/checkpoint.py", line 112, in create_from_checkpoint_aspect
    config_obj = config_class.parse_obj(config_as_dict)
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/pydantic/main.py", line 526, in parse_obj
    return cls(**obj)
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/pydantic/main.py", line 342, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for MySQLConfig
__root__
  Cannot specify both schema_pattern and database_pattern in the same config. Note that schema_pattern has been deprecated in favor of database_pattern. (type=value_error)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/logging/__init__.py", line 1025, in emit
    msg = self.format(record)
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/concurrent/futures/thread.py", line 80, in _worker
    work_item.run()
  File "/Users/jpharo/.pyenv/versions/3.7.13/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/cli/ingest_cli.py", line 134, in run_pipeline_to_completion
    pipeline.run()
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/run/pipeline.py", line 350, in run
    self.preview_workunits if self.preview_mode else None,
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/sql/sql_common.py", line 672, in get_workunits
    yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/state/stale_entity_removal_handler.py", line 250, in gen_removed_entity_workunits
    self.job_id, self.state_type_class
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/state/stateful_ingestion_base.py", line 263, in get_last_checkpoint
    job_id, checkpoint_state_class
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/state/stateful_ingestion_base.py", line 247, in _get_last_checkpoint
    state_class=checkpoint_state_class,
  File "/Users/jpharo/Code/datahub/images/venv/lib/python3.7/site-packages/datahub/ingestion/source/state/checkpoint.py", line 116, in create_from_checkpoint_aspect
    "Failed to construct checkpoint's config from checkpoint aspect.", e
Message: "Failed to construct checkpoint's config from checkpoint aspect."
Arguments: (ValidationError(model='MySQLConfig', errors=[{'loc': ('__root__',), 'msg': 'Cannot specify both schema_pattern and database_pattern in the same config. Note that schema_pattern has been deprecated in favor of database_pattern.', 'type': 'value_error'}]),)

```

**Recipe with redacted info:**

#### Recipe
```yaml
source:
  type: "mysql"
  config:
    host_port: "${MYSQL_HOST_PORT}"
    database: "${MYSQL_DB}"
    username: "datahub-ro-dev-001"
    password: "${DEV_PASSWORD}"
    database_alias: "[REDACTED]"
    database_pattern:
      allow:
        - "[REDACTED]"
    table_pattern:
        deny:
        - ".*__ptosc.*"
    stateful_ingestion:
        enabled: True

pipeline_name: "mysql_dev_pipeline"

sink:
  type: datahub-kafka
  config:
    connection:
      bootstrap: "[REDACTED]:9092"
      schema_registry_url: "[REDACTED]"
    topic_routes:
      MetadataChangeEvent: datahub_dev_MetadataChangeEvent_v4
      MetadataChangeProposal: datahub_dev_MetadataChangeProposal_v1

datahub_api:
  server: "[REDACTED]"
```

### GraphQL response 

I retrieved the datahubIngestionCheckpoint via the  `graph.get_latest_timeseries_value` in iPython to get a closer look at the server response. It seems even if I never ran ingestion with `schema_pattern` set in my recipe, it is returning both `schema_pattern` and `database_pattern` and blowing up when it tried to create a checkpoint instead from this response.

```
DatahubIngestionCheckpointClass({
  'timestampMillis': 1667604888191,
  'eventGranularity': None,
  'partitionSpec': PartitionSpecClass({
     'type': 'FULL_TABLE',
     'partition': 'FULL_TABLE_SNAPSHOT',
     timePartition': None
  }),
  'messageId': None,
  'pipelineName': 'mysql_dev_pipeline',
  'platformInstanceId': '[REDACTED]',
  'config': '{
    "env": "PROD",
    "platform": null,
    "platform_instance": null,
    "stateful_ingestion": {
      "enabled": true,
      "max_checkpoint_state_size": 16777216,
      "state_provider": {
        "type":
        "datahub",
        "config": null
     },
     "ignore_old_state": false,
     "ignore_new_state": false,
     "remove_stale_metadata": true,
     "fail_safe_threshold": 95.0
    },
    "options": {"echo": true},
    "schema_pattern": {"allow": [".*"], "deny": [], "ignoreCase": true},
    "table_pattern": {"allow": [".*"], "deny": [".*__ptosc.*"], "ignoreCase": true},
    "view_pattern": {"allow": [".*"], "deny": [".*__ptosc.*"], "ignoreCase": true},
    "profile_pattern": {"allow": [".*"], "deny": [], "ignoreCase": true},
    "domain": {},
    "include_views": true,
    "include_tables": true,
    "profiling": {
      "enabled": false,
      "limit": null,
      "offset": null,
      "report_dropped_profiles": false,
      "turn_off_expensive_profiling_metrics": false,
      "profile_table_level_only": false,
      "include_field_null_count": true,
      "include_field_min_value": true,
      "include_field_max_value": true,
      "include_field_mean_value": true,
      "include_field_median_value": true,
      "include_field_stddev_value": true,
      "include_field_quantiles": false,
      "include_field_distinct_value_frequencies": false,
      "include_field_histogram": false,
      "include_field_sample_values": true,
      "max_number_of_fields_to_profile": null,
      "profile_if_updated_since_days": 1,
      "profile_table_size_limit": 1,
      "profile_table_row_limit": 50000,
      "max_workers": 40,
      "query_combiner_enabled": true,
      "catch_exceptions": true,
      "partition_profiling_enabled": true,
      "bigquery_temp_table_schema": null,
      "partition_datetime": null
    },
    "username": "datahub-ro-dev-001",
    "host_port": "[REDACTED]",
    "database": "[REDACTED]",
    "database_alias": null,
    "scheme": "mysql+pymysql",
    "sqlalchemy_uri": null,
    "database_pattern": {"allow": [REDACTED], "deny": [], "ignoreCase": true}}',
    'state': IngestionCheckpointStateClass({
       'formatVersion': '1.0',
       'serde': 'base85',
       'payload': b'[REDACTED]'
    }),
  'runId': 'mysql-2022_11_04-15_34_42'
})
```
